#01-INSPECCIÓN INICIAL

In [10]:
#Carga del dataset
import pandas as pd

df = pd.read_json(r"C:\Users\jorge leandro\OneDrive\Desktop\proyecto_final_md\data\raw\streaming_users_dirty.json")

In [11]:
#Estructura del dataset
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8160 entries, 0 to 8159
Data columns (total 8 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   user_id                   8160 non-null   int64  
 1   age                       8160 non-null   int64  
 2   subscription_plan         8160 non-null   object 
 3   monthly_watch_time_mins   7967 non-null   float64
 4   country                   8160 non-null   object 
 5   favorite_genre            7920 non-null   object 
 6   last_login_date           7840 non-null   object 
 7   customer_support_tickets  8160 non-null   int64  
dtypes: float64(1), int64(3), object(4)
memory usage: 510.1+ KB


In [12]:
#Visualización de los primeros registros
df.head()


,user_id,age,subscription_plan,monthly_watch_time_mins,country,favorite_genre,last_login_date,customer_support_tickets
0,10000,39,Estándar,805.8,Brasil,Crime,2025-03-04,99
1,10001,37,Estándar,1173.4,Colombia,Crime,2019-04-02,2
2,10002,28,Básico,401.0,Colombia,Crime,2018-04-13,0
3,10003,43,Básico,62.4,Uruguay,Thriller,2021-01-31,0
4,10004,51,Básico,477.8,Perú,Thriller,2020-09-30,1


In [13]:
#Dimensiones del dataset
df.shape


(8160, 8)

In [14]:
#Tipos de datos
df.dtypes


user_id                       int64
age                           int64
subscription_plan            object
monthly_watch_time_mins     float64
country                      object
favorite_genre               object
last_login_date              object
customer_support_tickets      int64
dtype: object

In [15]:
# Conteo de valores nulos por columna
print("Cantidad de valores faltantes:")
print(df.isnull().sum())

Cantidad de valores faltantes:
user_id                       0
age                           0
subscription_plan             0
monthly_watch_time_mins     193
country                       0
favorite_genre              240
last_login_date             320
customer_support_tickets      0
dtype: int64


In [16]:
# Cantidad de registros completamente duplicados
print(f"Cantidad de filas duplicadas: {df.duplicated().sum()}")

Cantidad de filas duplicadas: 126


In [17]:
# Resumen estadístico de las variables numéricas
df.describe().round(2)

,user_id,age,monthly_watch_time_mins,customer_support_tickets
count,8160.00,8160.00,7967.00,8160.00
mean,13995.43,34.10,1107.35,1.80
std,2310.81,14.51,5310.44,11.33
min,10000.00,-5.00,-120.00,-1.00
25%,11987.75,25.00,489.20,0.00
50%,13998.50,33.00,757.40,1.00
75%,15997.25,42.00,1045.70,1.00
max,17999.00,150.00,99999.00,150.00


Este dataset representa:

Información de usuarios de una plataforma de streaming: edad, país, plan, hábitos de consumo, género favorito, actividad reciente y reclamos al soporte.

Cuenta con 8160 registros. 

Existen valores faltantes en las columnas last_login_date(320), montly_watch_time_mins (193) y favorite_genre (240); y 126 registros duplicados
Además de los valores nulos y duplicados, se pueden visualizar incoherencias en categoria edad (-5,150), tiempo mensual de visualización en minutos (-120),reclamos al soporte(-1); e inconsistencias en categorías plan de suscripción, país y género favorito en donde varía el formato de escritura.